api/controller.py

```diff
@@ -164,13 +163,26 @@ def __init__(self, _db, testing=False):
         self.load_settings()
 
     def load_facets_from_request(self, *args, **kwargs):
-        """Load a faceting object from the incoming request, but also apply an authentication
-        restriction.
+        """Load a faceting object from the incoming request, but also apply some
+        application-specific access restrictions:
 
-        Specifically, in this application you can't use nonstandard
-        caching rules unless you're an authenticated administrator.
+        * You can't use nonstandard caching rules unless you're an authenticated administrator.
+        * You can't access a WorkList that's not accessible to you.
```

Comment:
`load_facets_from_request` is called every time we're building an OPDS feed, and it already might short-circuit the request by returning a ProblemDetail (if the facets are incorrectly specified) so it's a convenient place to put other logic that might short-circuit the request by returning a ProblemDetail.